### PR TITLE
`Development`: Add documentation for parallel server test execution

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Server test documentation is currently missing an explanation of parallel test execution, making it difficult for developers to understand. This PR adds the corresponding section.

### Description

Added the section `7. Parallel test execution` to the server test guidelines. It explains how to enable parallel execution, how resources can be shared, and how a test can be run in isolation.

### Steps for Testing

Look at the docs [here](https://artemis-platform--7673.org.readthedocs.build/en/7673/dev/guidelines/server-tests.html).

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: docs/dev/guidelines/server-tests.rst [hunk 1]
```diff
@@ -178,5 +178,44 @@ We no longer mock the uppermost method but only throw the exception at the place
...
+===========================
+We use the `JUnit 5 <https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution>`__ feature to execute tests in parallel.
 
+The following line in the ``junit-platform.properties`` file enables parallel test execution with JUnit 5. Setting the property to ``false`` disables parallel test execution.
 
+.. code-block:: properties
+
+    junit.jupiter.execution.parallel.enabled = true
+
+To execute a test class and its inheriting classes in parallel, annotate it with ``@Execution(ExecutionMode.CONCURRENT)``. Since we need to isolate resources such as
```
Comment (b-fein) [inline docs/dev/guidelines/server-tests.rst:191]: The current reading flow is alternating

- ResourceLock
- Isolated
- ResourceLock
- Isolated.

This makes it somewhat hard to follow.

Could you maybe start a small section only for resource locks, and then another one for isolated? That would make the structure and intention more understandable:
- by default please use `@Execution(ExecutionMode.CONCURRENT)`
- if not possible, a resource lock still allows limited concurrency, and
- if that is not possible either, use `@Isolated`.

```suggestion
To execute a test class and its inheriting classes in parallel, annotate it with ``@Execution(ExecutionMode.CONCURRENT)``.

ResourceLock
''''''''''''
Since we need to isolate resources such as […]


Isolated
''''''''
[…]
```

Comment (dearjasmina) [reply docs/dev/guidelines/server-tests.rst:191]: I agree, maybe one addition would be to list example tests directly from Artemis so that people can look directly in the codebase to see the differences.

File: docs/dev/guidelines/server-tests.rst [hunk 2]
```diff
@@ -178,5 +178,44 @@ We no longer mock the uppermost method but only throw the exception at the place
...
+    @SpringBootTest
+    @AutoConfigureMockMvc
+    @ExtendWith(SpringExtension.class)
+    @Execution(ExecutionMode.CONCURRENT)
+    @ResourceLock("AbstractSpringIntegrationIndependentTest")
+    @AutoConfigureEmbeddedDatabase
+    @ActiveProfiles({ SPRING_PROFILE_TEST, "artemis", "scheduling", "lti" })
+    @TestPropertySource(properties = { "artemis.user-management.use-external=false" })
+    public abstract class AbstractSpringIntegrationIndependentTest extends AbstractArtemisIntegrationTest { ... }
+
```
[the remark this answers is not in the document]
Comment (laadvo) [reply docs/dev/guidelines/server-tests.rst:207]: This makes sense, it is now explained that the annotation should be added to the base class, and that the groups run in parallel whereas the tests within one group run sequentially

**Review by b-fein — APPROVED** [review]
Thanks!

**Review by dearjasmina — APPROVED** [review]
The documentation is really good, everything is well explained
